src/extension.ts

```diff
@@ -93,6 +93,22 @@ export async function activateInternal(context: vscode.ExtensionContext) {
     context.subscriptions.push(ext.outputChannel);
     vscode.commands.executeCommand('setContext', 'isEditorEnabled', false);
 
+    // Add XML schema association for policy files
+    const xmlExtension = vscode.extensions.getExtension('redhat.vscode-xml');
+    if (xmlExtension) {
+        if (!xmlExtension.isActive) {
+            await xmlExtension.activate();
+        }
+        // Get the XML API
+        const xmlAPI = xmlExtension.exports;
+        if (xmlAPI && xmlAPI.addXMLFileAssociations) {
+            xmlAPI.addXMLFileAssociations([{
+                pattern: "**/*.policy.xml",
+                systemId: context.asAbsolutePath("resources/policySchemas/policies.xsd")
+            }]);
+        }
```

Comment:
shall we ask for user to install this extension if no such extension? you may reference this code
https://github.com/microsoft/vscode-azureapicenter/blob/95e6d5a413aac9d75559667926f759f5355b485a/src/utils/ensureExtension.ts#L7

Reply:
Thanks, this extension is added as APIM extension's dependency so we can assume the extension exists.